lib/active_model/serializer.rb

```diff
@@ -60,16 +60,10 @@ class << self
 
     # @api private
     def self.serializer_lookup_chain_for(klass)
-      chain = []
-
-      resource_class_name = klass.name.demodulize
-      resource_namespace = klass.name.deconstantize
-      serializer_class_name = "#{resource_class_name}Serializer"
-
-      chain.push("#{name}::#{serializer_class_name}") if self != ActiveModel::Serializer
-      chain.push("#{resource_namespace}::#{serializer_class_name}")
-
-      chain
+      lookups = ActiveModelSerializers.config.serializer_lookup_chain
+      Array[*lookups].map do |lookup|
```

Comment:
oh ok. that would be a safe way to handle the accidental scenario of lookups being a nested array